src/server_manager/web_app/shadowbox_server.ts

```diff
@@ -125,6 +125,23 @@ export class ShadowboxServer implements server.Server {
     return new Date(this.serverConfig.createdTimestampMs);
   }
 
+  getHostname(): string {
+    try {
+      return new URL(this.managementApiAddress).hostname;
+    } catch (e) {
+      return '';
+    }
+  }
+
+  getManagementPort(): number {
+    try {
+      // Convert URL port string to number
+      return +(new URL(this.managementApiAddress).port);
```

Comment:
Is there a more obvious way of converting to a number?

Reply:
Changed to `parseInt` in 117fcac.